_posts/en/newsletters/2023-10-11-newsletter.md

```diff
@@ -0,0 +1,189 @@
+by Niklas Gögge (dergoegge) that improves type safety by introducing
+separate types for `txid` (the transaction identifier or hash that
+doesn't include the segwit witness data) and `wtxid` (same but includes
+the witness data), rather than both being represented by a `uint256`
+(a 256-bit integer, which can contain a SHA256 hash). This PR
+should have no operational effect; it aims to prevent future
+programming errors in which one kind of transaction ID is used where
+the other was intended. Such errors will be detected at compile time.
+
+To minimize disruption and ease review, these new types will initially
+be used in only one area of the code (the transaction "orphanage");
+future PRs will deploy the new types to other areas of the codebase.
+
+{% include functions/details-list.md
+  q0="What does it mean for a transaction identifier to be type-safe?
+      Why is that important or helpful? Are there any downsides?"
+  a0="A transaction identifier is considered type-safe when it undergoes
+      rigorous type-checking during compile-time, guaranteeing that
+      operations involving these identifiers are thoroughly validated
+      and constrained to their intended uses before the program is
+      executed, thus preventing type-related errors and ensuring
+      program reliability. No downsides were noted."
+  a0link="https://bitcoincore.reviews/28107#l-38"
+
+  q1="Rather than the new class types `Txid` and a `Wtxid` _inheriting_
+      from `uint256`, should they  _include_ (wrap) a `uint256`?
+      What are the tradeoffs?"
+  a1="Those classes could do that, but it would cause much more code
+      churn (many more source lines would need to be touched)."
+  a1link="https://bitcoincore.reviews/28107#l-43"
+
+  q2="Why is it better to enforce types at compile-time instead of at
+      run-time?"
+  a2="Developers discover errors quickly as they're coding, rather
+      than relying on writing extensive test suites to catch bugs in runtime
+      (and these tests may still miss some errors). However, testing would
+      still be useful since type safety won't prevent consistent use of the
+      wrong type of transaction ID (but consistently) in the first place."
+  a2link="https://bitcoincore.reviews/28107#l-67"
+
+  q3="Conceptually, when writing new code that requires referring to
+      transactions, when should you use `txid` and when should you use
+      `wtxid`? Can you point to any examples in the code where using one
+      instead of the other could be very bad?"
+  a3="In general, use of `wtxid` is preferred since it commits to the
+      entire transaction. An important exception is the `prevout`
+      reference from each input to the output (UTXO) it's spending.
+      It must be a `txid` to retain compatibility with pre-segwit nodes.
+      (Note that it's not possible for only upgraded segwit nodes to use
+      `wtxid` here, because this would change the `txid` of the spending
+      transaction.)
```

Comment:
"compatibility" suggests to me that there is another way of referencing prevouts in post-segwit nodes.
The note doesn't really make sense to me - why would segwit nodes reincorporate witnesses into prevouts, seems to defeat the purpose/definition of segwit?
```suggestion
      entire transaction. An important exception is the `prevout`
      reference from each input to the output (UTXO) it's spending,
      which specify transactions by txid.
```